### PR TITLE
Build rubies for ubuntu-24.04

### PR DESCRIPTION
**Pull request by kachick**
For a preparation part of https://github.com/ruby/setup-ruby/issues/595

But please close this PR if the automated bot covers these OS updates. 🙇‍♂️

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ ubuntu-20.04, ubuntu-22.04, macos-11, macos-arm-oss ]
+        os: [ ubuntu-20.04, ubuntu-22.04, ubuntu-24.04, macos-11, macos-arm-oss ]
         ruby: [jruby-9.4.7.0]
```
Comment (kachick) [inline .github/workflows/build.yml:14]: Should I add other rubies in this matrix for this case?

Comment (eregon) [reply .github/workflows/build.yml:14]: I typically do it with 1 commit per major.minor, you can see that in existing commits.
IIRC there is some script to easily get such list in this repo or maybe in setup-ruby.
Unfortunately this procedure is not very friendly for PRs.
You could update this one to build all 3.3 releases though, only on 24.04 and if: false on the jruby job below

**Comment by eregon** [comment]
I started the builds with https://github.com/ruby/ruby-builder/commit/208503aa5130913709c10e03fa296c9b19ab7a9c

**Comment by kachick** [comment]
👍

**Pull request closed without merging**